server/controllers/request.js

```diff
@@ -0,0 +1,81 @@
+const Request = require("../models/Request");
+const User = require("../models/User");
+const asyncHandler = require("express-async-handler");
+
+// @route POST /request
+// @desc create a new pet sitting request
+// @access Private
+exports.makeRequest = asyncHandler(async (req, res) => {
+  const body = req.body;
+  if (!(body || body.sitterId || body.ownerId || body.duration)) {
+    res.status(400);
+    throw new Error("Bad Request");
+  }
+  const sitter = await User.findById(body.sitterId);
+  if (!sitter) {
+    res.status(404);
+    throw new Error("Sitter doesn't exist");
+  }
+  const newRequest = await Request.create({
+    ownerId: req.user.id,
+    sitterId: body.sitterId,
+    duration: {
+      start: new Date(body.duration.start),
+      end: new Date(body.duration.end),
+    },
+  });
+  if (newRequest) {
+    res.status(200).json({
+      success: {
+        request: newRequest,
+      },
+    });
+  } else {
+    res.status(500);
+    throw new Error("Error creating request");
+  }
+});
+
+// @route GET /request
+// @desc Get requests for a logged in user
+// @access Private
+exports.getRequest = asyncHandler(async (req, res) => {
+  const requests = await Request.find({ sitterId: req.user.id });
```

Comment:
Why query the database to obtain requests sent which belong to sitters too? 

Eg: `Request.find({ $or: [{ ownerId: userId }, { sitterId: userId }] })`

Reply:
Change made. I figured our current request page was only for sitters so I initially implemented the controller as such.